### ISSUE TITLE
Google Cloud Provider attributes vs. modules usage in 0.4.0

### ISSUE THREAD

**Issue by jrslv** [hashicorp/terraform#1375]
In our project https://github.com/ContainerSolutions/terraform-mesos we have the following provider.tf
```
provider "google" {
    account_file = "${var.account_file}"
    project = "${var.project}"
    region = "${var.region}"
}
```
When we use this repository as a module:
```
module "mesos" {
    source                      = "github.com/ContainerSolutions/terraform-mesos"
    account_file                = "/Users/jrslv/.terraform/ContainerSolutions-b4110aa1ff16.json"
    project                     = "container-solutions"
    region                      = "europe-west1"
    zone                        = "europe-west1-d"
    gce_ssh_user                = "jrslv"
    gce_ssh_private_key_file    = "/Users/jrslv/.ssh/google_compute_engine"
    name                        = "jrslv"
    masters                     = "3"
    slaves                      = "3"
    network                     = "10.20.30.0/24"
    localaddress                = "92.111.228.8/32"
    domain                      = "example.com"
}
```

then `terraform plan` returns: 

```
Errors:

  * provider.google: "project": required field is not set
  * provider.google: "region": required field is not set
```

If we use .tfvars in the project, it works. If we use terraform 0.3.7, it works. If we run `terraform apply`, it asks for `account_file`, `project` and `region` and works just fine.

We even tried to hardcode the values to `provider.tf`, but it did not work, so the issue probably isn't variables passing to the module, but a combination of modules usage and the Google Cloud provider.

Not even passing the variables like `terraform plan -var 'project=container-solutions' -var 'region=europe-west1'` doesn't work. Maybe the `terraform plan` command is a bit broken.

**Comment by phinze** [comment]
Hi there - thanks for the report! I'm almost positive this is the same issue as #1380, which I happened to pick up first, but I'll leave this open until I get a chance to verify.

If it is the same issue, you should be able to work around it for now by using the `google-cloud` provider's env vars for configuration: `GOOGLE_ACCOUNT_FILE`, `GOOGLE_PROJECT`, and `GOOGLE_REGION`.

I'll keep you posted as I make progress on tracking this down. :ok_hand: :bug:

**Comment by thijsschnitger** [comment]
I can confirm it works when using env vars. One thing I found, it complains about the project and the region when I don't set them, but not the account_file. Terraform will ask for the account_file when I do a terraform plan.

**Comment by phinze** [comment]
This should be fixed in 0.4.2 - feel free to follow up if you still see issues. :+1:
